### PR TITLE
[v2] Increase metrics counter in v2 responses.

### PR DESCRIPTION
**Pull request by simitt**
implements elastic/apm-server#1355

### CODE REVIEW

File: beater/v2_handler.go [hunk 1]
```diff
@@ -21,53 +21,65 @@ import (
...
 
-	"github.com/pkg/errors"
-
 	"github.com/elastic/apm-server/decoder"
 )
 
-var (
-	errInvalidMetadataFormat = errors.New("invalid metadata format")
-)
+const MethodNotAllowedErrType stream.StreamError = 100
```
Comment (graphaelli) [inline beater/v2_handler.go]: any reason this isn't at the bottom of the list in `processor/stream/result.go` ?

File: beater/v2_handler.go [hunk 2]
```diff
@@ -21,53 +21,65 @@ import (
...
 func (v *v2Handler) sendResponse(logger *logp.Logger, w http.ResponseWriter, sr *stream.Result) {
-	statusCode := v.statusCode(sr)
+	statusCode, counter := v.statusCode(sr)
+	responseCounter.Inc()
+	counter.Inc()
 
 	w.WriteHeader(statusCode)
 	if statusCode != http.StatusAccepted {
+		responseErrors.Inc()
 		// this singals to the client that we're closing the connection
```
Comment (graphaelli) [inline beater/v2_handler.go]: might as well fix this typo in `singals`

File: beater/v2_handler.go (hunk 1 above)
Comment (simitt) [reply beater/v2_handler.go]: `processor/stream` has no knowledge about the used protocol, it is just reading and processing a stream. The `MethodNotAllowedErrType` reflects that the wrong http method was used.
But I guess the err name is generic enough to move it inside `processor` as it doesn't say anything about the protocol.

**Comment by simitt** [comment]
@graphaelli I think it is fair to only count the error corresponding to the returned error code for `error.response`. 
As I share the uncertainty about generally only reporting this error, I introduced `apm-server.processor.stream.errors.*` and `apm-server.processor.stream.accepted` counters, counting all processed events.

**Comment by simitt** [comment]
@graphaelli retriggering jenkins solved the ci issue. It was some issue with `go vet` and not a flaky test, so tests should be fine.

**Pull request merged**